src/m365/planner/commands/bucket/bucket-get.ts

```diff
@@ -0,0 +1,215 @@
+}
+
+class PlannerBucketGetCommand extends GraphCommand {
+  public get name(): string {
+    return commands.BUCKET_GET;
+  }
+
+  public get description(): string {
+    return 'Gets the Microsoft Planner bucket in a plan';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.id = typeof args.options.id !== 'undefined';
+    telemetryProps.title = typeof args.options.title !== 'undefined';
+    telemetryProps.planId = typeof args.options.planId !== 'undefined';
+    telemetryProps.planName = typeof args.options.planName !== 'undefined';
+    telemetryProps.ownerGroupId = typeof args.options.ownerGroupId !== 'undefined';
+    telemetryProps.ownerGroupName = typeof args.options.ownerGroupName !== 'undefined';
+    return telemetryProps;
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['id', 'name', 'planId', 'orderHint'];
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    this
+      .getBucketId(args)
+      .then((bucketId: string) => this.getBucketById(bucketId))
+      .then((bucket: PlannerBucket) => {
+        logger.log(bucket);
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  private getBucketId (args: CommandArgs): Promise<string> {
+    if (args.options.id) {
+      return Promise.resolve(args.options.id);
+    }
+
+    return this
+      .getPlanId(args)
+      .then((planId: string) => {
+        const requestOptions: any = {
+          url: `${this.resource}/v1.0/planner/plans/${planId}/buckets`,
+          headers: {
+            accept: 'application/json;odata.metadata=none'
+          },
+          responseType: 'json'
+        };
+
+        return request.get<{ value: { id: string; name: string; }[] }>(requestOptions);
+      })
+      .then(response => {
+        const bucket: { id: string; name: string; } | undefined = response.value.find(val => val.name === args.options.title);
+
+        if (!bucket) {
+          return Promise.reject(`The specified bucket does not exist`);
+        }
+
+        return Promise.resolve(bucket.id);
+      });     
+  }
+
+  private getPlanId(args: CommandArgs): Promise<string> {
+    if (args.options.planId) {
+      return Promise.resolve(args.options.planId);
+    }
+
+    return this
+      .getGroupId(args)
+      .then((groupId: string) => {
+        const requestOptions: any = {
+          url: `${this.resource}/v1.0/planner/plans?$filter=(owner eq '${groupId}')`,
+          headers: {
+            accept: 'application/json;odata.metadata=none'
+          },
+          responseType: 'json'
+        };
+
+        return request.get<{ value: { id: string; title: string; }[] }>(requestOptions);
+      })
+      .then(response => {
+        const plan: { id: string; title: string; } | undefined = response.value.find(val => val.title === args.options.planName);
```

Comment:
Please also handle if there are multiple plans. Return a message that they should choose from an id. You can find an example how this is done with Buckets here: https://github.com/pnp/cli-microsoft365/blob/f028e14fa587213b1868cdee83f77b7c99d109ce/src/m365/planner/commands/bucket/bucket-remove.ts#L114-L124

Reply:
Good suggestion:)